src/plone/api/portal.py

```diff
@@ -431,3 +434,41 @@ def translate(msgid, domain="plone", lang=None):
         # Pass the request, so zope.i18n.translate can negotiate the language.
         query["context"] = getRequest()
     return translation_service.utranslate(**query)
+
+
+@required_parameters("name")
+def get_vocabulary(name=None, context=None):
+    """Return a vocabulary object with the given name.
+
+    :param name: Name of the vocabulary.
+    :type name: str
+    :param context: Context to be applied to the vocabulary. Default: portal root.
+    :type context: object
+    :returns: A SimpleVocabulary instance that implements IVocabularyTokenized.
+    :rtype: zope.schema.vocabulary.SimpleVocabulary
+    :Example: :ref:`portal-get-vocabulary-example`
+    """
+    if not context:
+        context = get()
+    try:
+        vocabulary = getUtility(IVocabularyFactory, name)
+    except ComponentLookupError:
+        raise InvalidParameterError(
+            "Cannot find a vocabulary with name '{name}'.\n"
+            "Available vocabularies are:\n"
+            "{vocabularies}".format(
+                name=name,
+                vocabularies="\n".join(get_vocabularies_names()),
+            ),
+        )
+    return vocabulary(context)
+
+
+def get_vocabularies_names():
```

Comment:
@stevepiercy is this more correct?
```suggestion
def get_vocabulary_names():
```

Reply:
Grammatically in English, it would be "get vocabularies' names" which has the same meaning as "get the names of the vocabularies", but we can't use spaces or `'`, so this is fine as is.

Reply:
@stevepiercy is technically correct so I can't justify this in terms of grammar rules, but `get_vocabulary_names` sounds much less awkward to me than `get_vocabularies_names`.

Reply:
I also had the feeling that `get_vocabulary_names` is better.
FWIW I also tried some prompts on https://lmarena.ai/ and it seems they agree with `get_vocabulary_names`.

Reply:
okay so `vocabularies` => `vocabulary_names` everywhere?

Reply:
> okay so `vocabularies` => `vocabulary_names` everywhere?

I think it is rather `vocabularies_names` => `vocabulary_names` everywhere, but let's wait for @stevepiercy reply.

Reply:
`get_vocabulary_names` is OK, too.